test/unit/alphabet/semi_alphabet_constexpr_test_template.hpp

```diff
@@ -0,0 +1,156 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alphabet/concept.hpp>
+
+using namespace seqan3;
+
+template <typename T>
+using semi_alphabet_constexpr = ::testing::Test;
+
+TYPED_TEST_CASE_P(semi_alphabet_constexpr);
+
+TYPED_TEST_P(semi_alphabet_constexpr, concept_check)
+{
+    EXPECT_TRUE(detail::constexpr_semialphabet<TypeParam   >);
+    EXPECT_TRUE(detail::constexpr_semialphabet<TypeParam & >);
+
+    EXPECT_TRUE(detail::constexpr_semialphabet<TypeParam const   >);
+    EXPECT_TRUE(detail::constexpr_semialphabet<TypeParam const & >);
+
+    EXPECT_TRUE(detail::writable_constexpr_semialphabet<TypeParam   >);
+    EXPECT_TRUE(detail::writable_constexpr_semialphabet<TypeParam & >);
+
+    EXPECT_FALSE(detail::writable_constexpr_semialphabet<TypeParam const   >);
+    EXPECT_FALSE(detail::writable_constexpr_semialphabet<TypeParam const & >);
+}
+
+TYPED_TEST_P(semi_alphabet_constexpr, default_value_constructor)
+{
+    [[maybe_unused]] constexpr TypeParam t0{};
+}
+
+TYPED_TEST_P(semi_alphabet_constexpr, copy_constructor)
+{
+    constexpr TypeParam t1{};
```

Comment:
Maybe with an assigned value? I think the test has then more value

Reply:
These tests are copy and pasted from the alphabet_test_template. Changing them is not in the scope of this PR I think

Reply:
But what is the value of this test. If I have written a shitty copy constructor that in fact just calls he default constructor I won't be able to see anything. I think that should be covered. And also corrected from the source of this template if it is wrong there as well.

Reply:
> But what is the value of this test. If I have written a shitty copy constructor that in fact just calls he default constructor I won't be able to see anything. I think that should be covered.

So you are suggesting that the variable should hold a value != the default? 
I see your point but it is also a little against the unit test design because in order to (generically) assign a different value than the default I would need to call `assign_rank_to`. So I would not know whether the copy construction failed or the assign_rank_to function? 

> And also corrected from the source of this template if it is wrong there as well.

Sorry, I meant "this was moved here from the alphabet_test_template". So this is the only place.

Reply:
I changed them.

> And also corrected from the source of this template if it is wrong there as well.

Sorry, I meant "there were **moved** here from the former alphabet_test_template", so there is only one place to change them.